### PR TITLE
Feat/miaw dns zone

### PR DESCRIPTION
**Pull request by marie-j**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-4025
| License          | BSD 3-Clause

## Description
 Order standalone DNS Zone via Agora

### CODE REVIEW

File: packages/manager/modules/product-offers/src/workflows/product-offers-order-workflow.class.js [hunk 1]
```diff
@@ -76,13 +74,12 @@ export default class OrderWorkflow extends Workflow {
...
   /**
-   * Get the pricings from the catalog, which match the plan code, which is get
-   * by linked method 'onBeforePricingGetPlancode'.
+   * Get the pricings from the catalog matching the planCode
    * @return {Promise} Promise of the catalog pricings
    */
   getPricings() {
     this.pricing = null;
-    this.planCode = this.onBeforePricingGetPlancode();
+    // this.planCode = this.onBeforePricingGetPlancode();
```
Comment (antleblanc) [inline packages/manager/modules/product-offers/src/workflows/product-offers-order-workflow.class.js:82]: Could be removed then:)

File: packages/manager/apps/web/client/app/dns-zone/new/translations/Messages_fr_FR.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{
+  "domains_newdnszone_order_title": "Ajouter une zone DNS",
+  "domains_newdnszone_order_step1_title": "Entrez un nom de domaine",
+  "domains_newdnszone_order_step1_zonename": "Nom de domaine",
+  "domains_newdnszone_order_step1_zonename_pending": "Vérification de la disponibilité du nom de la zone...",
+  "domains_newdnszone_order_step2_title": "Détails",
+  "domains_newdnszone_order_step2_minimized": "Activer les entrées minimales?",
```
Comment (antleblanc) [inline packages/manager/apps/web/client/app/dns-zone/new/translations/Messages_fr_FR.json:7]:
```suggestion
  "domains_newdnszone_order_step2_minimized": "Activer les entrées minimales ?",
```

File: packages/manager/apps/web/client/app/dns-zone/new/dns-zone-new.html [hunk 3]
```diff
@@ -1,313 +1,137 @@
...
+    </oui-message>
 
-            <!-- step 1 -->
-            <h2
-                class="mt-5"
-                data-translate="domains_newdnszone_order_step1_title"
-            ></h2>
-            <form name="newDnsZoneForm">
+    <ovh-manager-product-offers
+        workflow-type="'order'"
```
Comment (JeremyDec) [inline packages/manager/apps/web/client/app/dns-zone/new/dns-zone-new.html:23]: You should use this constant: https://github.com/ovh/manager/blob/master/packages/manager/modules/product-offers/src/workflows/product-offers-workflow.constants.js#L7

File: packages/manager/apps/web/client/app/dns-zone/new/dns-zone-new.html [hunk 4]
```diff
@@ -1,313 +1,137 @@
...
+        workflow-options="{
+            catalog: $ctrl.catalog,
+            catalogItemTypeName: 'plans',
```
Comment (JeremyDec) [inline packages/manager/apps/web/client/app/dns-zone/new/dns-zone-new.html:26]: Same here: https://github.com/ovh/manager/blob/master/packages/manager/modules/product-offers/src/workflows/product-offers-workflow.constants.js#L3

File: packages/manager/apps/web/client/app/dns-zone/new/dns-zone-new.html [hunk 5]
```diff
@@ -1,313 +1,137 @@
...
+            productName: 'dns',
+            planCode: 'zone',
+        }"
+        get-workflow-configuration="$ctrl.getConfiguration()"
+        pricing-type="'renew'"
```
Comment (JeremyDec) [inline packages/manager/apps/web/client/app/dns-zone/new/dns-zone-new.html:31]: And also: https://github.com/ovh/manager/blob/master/packages/manager/modules/product-offers/src/pricing/pricing.constants.js :)

File: packages/manager/modules/product-offers/src/product-offers.component.js [hunk 6]
```diff
@@ -70,6 +70,12 @@ export default {
      */
     workflowOptions: '<',
 
+    /**
+     * Getter for workflow configuration
+     * @type {function}
+     */
+    getWorkflowConfiguration: '&?',
```
Comment (JeremyDec) [inline packages/manager/modules/product-offers/src/product-offers.component.js:77]: Hmm why having this binding at this level? Order workflow is supposed to have its own `getConfiguration`method.
If you look here: https://github.com/ovh/manager/blob/master/packages/manager/modules/product-offers/src/workflows/product-offers-order-workflow.class.js#L30

Comment (marie-j) [reply packages/manager/modules/product-offers/src/product-offers.component.js:77]: The issue is we cannot bind to the current controller. If I set `onGetConfiguration: $ctrl.getConfiguration.bind($ctrl)` (to be able to use `this` inside the `getConfiguration` function) it triggers `digest` errors

Comment (marie-j) [reply packages/manager/modules/product-offers/src/product-offers.component.js:77]: I was having issue with binding `this` in the `getConfiguration` but now I've thought about it with your comment I might have found a solution (I'll update if this suit the needs)

Comment (JeremyDec) [reply packages/manager/modules/product-offers/src/product-offers.component.js]: Ok yes that's what I had in mind :)

**Pull request merged**